src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenConstructorInitTests.cs

```diff
@@ -592,6 +594,58 @@ .maxstack  4
 ");
         }
 
+        [WorkItem(42985, "https://github.com/dotnet/roslyn/issues/42985")]
+        [Fact]
+        public void DecimalConstInit002()
+        {
+            var source = @"
+class C
+{
+    const decimal d1 = 0.1m;
```

Comment:
>0.1m [](start = 23, length = 4)

Let's also add a test with ```0m```. Will it be treated as default and eliminate the constructor?

Reply:
I found that `0m` was treated differently than `0` or `default`. That seems like a bug.

Reply:
(actually, it seems like the precision of `0.0m` made it a non-default value. `0m` is treated as expected.)